numpy/random/mtrand/generate_randint_helpers.py

```diff
@@ -0,0 +1,152 @@
+"""
+Helper file for generating the necessary helper functions
+for each `dtype` in `np.random.randint`. The function
+`gen_randint_helpers` is called right before Cythonization
+in NumPy setup.
+
+"""
+
+# npy_dt : Desired C API data type
+# npy_udt : Unsigned version of desired C
+#           API data type (could be identical)
+# npy_dt : Desired Python API data type
+
+template = """
+def _rand_{npy_dt}(low, high, size, rngstate):
+    \"\"\"
+    _rand_{npy_dt}(low, high, size, rngstate)
+
+    Return an array of random np.{np_dt} integers between ``lo`` and ``hi``
+    inclusive for each ``lo``, ``hi`` pair in ``low`` and ``high``, where
```

Comment:
That was an option I had considered, but it received push-back in my earlier attempt to add `dtype` to `randint` (see <a href="https://github.com/numpy/numpy/pull/6869#issuecomment-167081108">here</a>).  It certainly remains an option, but I think given the initial resistance, it would need stronger approval before I tried it again, especially given the acrobatics you would have to go through to get it to work.